src/extension/providers/commonProvider/index.ts

```diff
@@ -0,0 +1,43 @@
+import { EventEmitter } from "events";
+import { postMessage } from "~/extension/providers/postMessage";
+import { PromiseQueue } from "~/extension/providers/promiseQueue";
+
+export default class CommonProvider {
```

Comment:
I think `ProviderBase` would be a better name. Also maybe move it to the `src/extensions/providers/` directory, because it's not really a provider in that sense but just a base class.